README.md

```diff
@@ -102,7 +102,52 @@ bcrypt.compare(someOtherPlaintextPassword, hash, function(err, res) {
     // res == false
 });
 ```
+### with promises
 
+bcrypt uses whatever Promise implementation is available in `global.Promise`. NodeJS >= 0.12 has a native Promise implementation built in. However, this should work in any Promises/A+ compilant implementation.
+
+```javascript
+var bcrypt = require('bcrypt');
+const saltRounds = 10;
+const myPlaintextPassword = 's0/\/\P4$$w0rD';
+const someOtherPlaintextPassword = 'not_bacon';
+```
+
+#### To hash a password:
```

Comment:
I think you can avoid doing many examples and just say something along the lines of: any async method which takes a callback, can optionally return a promise if no callback is provided (or something of that nature). And maybe do a quick few line example. No need to go over all the api methods again.